bin/kano-settings-onboot

```diff
@@ -429,13 +491,21 @@ if __name__ == '__main__':
         set_to_HDMI(False)
 
     # calculate changes needed
-    changes = compare_and_set_mode() or \
-        compare_and_set_full_range() or \
-        compare_and_set_overscan() or \
-        compare_and_set_optimal_resolution()
+    overrall_changes = False
 
-    if changes:
+    changes = compare_and_set_mode()
+    overrall_changes = overrall_changes or changes
+
+    changes = compare_and_set_full_range()
+    overrall_changes = overrall_changes or changes
 
+    changes = compare_and_set_overscan()
+    overrall_changes = overrall_changes or changes
+
+    changes = set_resolution_for_screen_kit() or compare_and_set_optimal_resolution()
+    overrall_changes = overrall_changes or changes
+
+    if changes:
```

Comment:
Should this be `overall_changes`? If so, presumably the above lines can be reduced to

```python
overall_changes = \
    compare_and_set_mode() \
    or compare_and_set_full_range()
    or changes = compare_and_set_overscan()
    or set_resolution_for_screen_kit() \
    or compare_and_set_optimal_resolution()
```

Hmm, except if we actually require many of them to make changes in which case it might stop on the first call returning `True`. I don't know what behaviour you actually want but either way it seems like you can completely remove the `changes` (or `overall_changes`) variable and simplify above.